pkg/pgmodel/ingestor/trace/writer.go

```diff
@@ -56,14 +60,25 @@ type traceWriterImpl struct {
 	tagCache     *clockcache.Cache
 }
 
+var register sync.Once
+
 func NewWriter(conn pgxconn.PgxConn) *traceWriterImpl {
-	return &traceWriterImpl{
+	writer := &traceWriterImpl{
 		conn:         conn,
 		schemaCache:  newSchemaCache(),
 		instLibCache: newInstrumentationLibraryCache(),
 		opCache:      newOperationCache(),
 		tagCache:     newTagCache(),
 	}
+	register.Do(func() {
```

Comment:
I don't think this is the right place for registering cache metrics. It should go in cache.go, when the cache is created.